selfdrive/car/toyota/interface.py

```diff
@@ -204,7 +204,7 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
     ret.enableBsm = 0x3F6 in fingerprint[0] and candidate in TSS2_CAR
 
     # Detect smartDSU, which intercepts ACC_CMD from the DSU (or radar) allowing openpilot to send it
-    if 0x2FF in fingerprint[0]:
+    if 0x2FF in fingerprint[0] or 0x2AA in fingerprint[0]:
```

Comment:
is this a different device from the smart DSU? the comment says it already works with the radar

Reply:
It's a different hardware/firmware with a lot more redundancy initially but the PR was rejected. #24695

Reply:
This version is also cheaper as it is not based on a panda.

Reply:
can you use the SDSU on these cars? this hardware just enables more features like changing ACC_TYPE?

Reply:
> can you use the SDSU on these cars? this hardware just enables more features like changing ACC_TYPE?

technically, yea, if you can find the right connector / waterproofing for it.

0x2AA is a copy of the stock 0x343 signal, which also passes through various stock ACC-related states like distance button, lead state, permit braking, etc.

In the original PR for opendbc:
`BO_ 682 ACC_CONTROL_COPY: 8 XXX`

Reply:
Let me explain why I didn't use SDSU and try to create this device(firmware) at the first place. 

1. I can get find a connector for the can gateway so I have to install this device in the engine bay. So it has to be small in size, water proof, easy wiring. 

2. The board is the well know MB CAN filter, a mileage manipulator. Which is very easy to get on eBay,Aliexpress. It has two CAN buses, 12V input. Perfect for our purpose and easy to wire. 

3. One of the feature which is vey important is update FW over can bus. 


> can you use the SDSU on these cars? this hardware just enables more features like changing ACC_TYPE?

Reply:
We have 0x2FF in the DBC, what does 0x2AA publish?

Reply:
> `BO_ 682 ACC_CONTROL_COPY: 8 XXX`

0x2AA is a copy of 0x343